docs/OnClusterBuildInstructions.md

```diff
@@ -0,0 +1,345 @@
+# On-Cluster OS Image Builds
+
+## Prerequisites
+
+-   OpenShift Cluster w/ 4.14.
+
+-   Admin privileges on the aforementioned cluster
+
+-   An image registry that one has push and pull permissions to, which will be
+    used for storing the final OS image. Pull permissions for this image registry
+    must be added to the global pull secret. See [<u>these
+    instructions</u>](https://docs.openshift.com/container-platform/4.13/openshift_images/managing_images/using-image-pull-secrets.html#images-update-global-pull-secret_using-image-pull-secrets)
+    for how to do this. One can also use an OpenShift ImageStream within the MCO namespace for this purpose.
+
+## Initial Setup
+
+1.  Obtain the following information:
+
+    1.  _(Optional)_ Create an OpenShift ImageStream in the MCO namespace:
+        ```console
+        $ oc create imagestream os-image -n openshift-machine-config-operator
+        ```
+
+    2.  The name of a pull secret for the base OS image and the
+        extensions container. This must be created as a Secret within
+        the MCO namespace. One can clone the global pull secret, if
+        desired: (`$ oc create secret docker-registry
+        global-pull-secret-copy -n openshift-machine-config-operator
+        --from-file=.dockerconfigjson=<(oc get secret/pull-secret
+        -n openshift-config -o json | jq -r
+        '.data\[".dockerconfigjson"\] | @base64d'`).
+
+    3.  The final image pullspec. For example:
+        `quay.io/myorg/myrepo:latest`. Any provided tags will be
+        ignored. Instead, images will be tagged with the name of the
+        rendered MachineConfig that was used to produce the image. For
+        example:
+        `quay.io/myorg/myrepo:rendered-worker-dd37180cefa6a1e88834966330c0c028`. If using an OpenShift Imagestream for this purpose, image pullspec may be obtained thusly:
+        ```console
+        $ oc get imagestream/os-image -n openshift-machine-config-operator -o=jsonpath='{.status.dockerImageRepository}'
+        image-registry.openshift-image-registry.svc:5000/openshift-machine-config-operator/os-image
+        ```
+
+    4.  The name of a push secret that will enable the final OS image to
+        be pushed to the above pullspec. This must be created as a
+        Secret within the MCO namespace. If using an OpenShift ImageStream, one can use the builder secret for the MCO namespace. Its name may be obtained thusly:
+        ```console
+        $ oc get secrets -n openshift-machine-config-operator -o name | grep "builder-dockercfg"
```

Comment:
Maybe we can skip '$' because in preview mode, when you copy the text using available icon it needs to be manually removed